src/test/java/io/strimzi/kafka/bridge/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,252 @@
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.mqtt;
+
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.handler.codec.mqtt.MqttQoS;
+import io.strimzi.kafka.bridge.mqtt.config.BridgeConfig;
+import io.strimzi.kafka.bridge.mqtt.config.KafkaConfig;
+import io.strimzi.kafka.bridge.mqtt.config.MqttConfig;
+import io.strimzi.kafka.bridge.mqtt.core.MqttServer;
+import io.strimzi.kafka.bridge.mqtt.mapper.MappingRulesLoader;
+import io.strimzi.test.container.StrimziKafkaContainer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.eclipse.paho.client.mqttv3.IMqttToken;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttAsyncClient;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Integration test for MQTT bridge
+ */
+public class MqttBridgetIT {
+
+    private static final Logger logger = LoggerFactory.getLogger(MqttBridgetIT.class);
+    private static final String MQTT_SERVER_HOST = "localhost";
+    private static final int MQTT_SERVER_PORT = 1883;
+    private static final String MQTT_SERVER_URI = "tcp://" + MQTT_SERVER_HOST + ":" + MQTT_SERVER_PORT;
+    private static final String KAFKA_TOPIC = "devices_bluetooth_data";
+    private static KafkaConsumer<String, String> kafkaConsumerClient = null;
+    private static MqttServer mqttBridge;
+    private static Thread prepareServerThread;
+    private static StrimziKafkaContainer kafkaContainer;
+    private String mqttClientId;
+
+
+    /**
+     * Start the kafka Cluster
+     * Start the MQTT bridge before all tests
+     */
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        final Map<String, Object> config = new HashMap<>();
+        String kafkaBootstrapServers = "localhost:9092";
```

Comment:
I am not sure this default initialization is really needed. If the Kafka container doesn't start (and we just log an exception) you are expecting to connect a "localhost:9092" which actually doesn't exist, while the test has to fail earlier I would say.

Reply:
good catch! I will just throw the exception.